packages/@lwc/features/src/runtime.ts

```diff
@@ -4,29 +4,37 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { FeatureFlag } from './flags';
-import { assert, create, isFalse, isTrue } from '@lwc/shared';
+import features, { FeatureFlagLookup, FeatureFlagValue } from './flags';
+import { assert, create, isFalse, isTrue, isUndefined } from '@lwc/shared';
 
-interface FeatureFlagConfig {
-    [name: string]: FeatureFlag;
-}
-
-const runtimeFlags: FeatureFlagConfig = create(null);
+const runtimeFlags: FeatureFlagLookup = create(null);
 
-function setFeatureFlag(name: string, value: FeatureFlag) {
+// This function is not whitelisted for use within components and is meant for
+// configuring runtime feature flags during app initialization.
+function setFeatureFlag(name: string, value: FeatureFlagValue) {
     assert.invariant(
         isTrue(value) || isFalse(value),
-        `Runtime feature flags must be set to a boolean value but received a(n) ${typeof value} value for the '${name}' flag.`
+        `Invalid ${typeof value} value specified for the "${name}" flag. Runtime feature flags can only be set to a boolean value.`
     );
+    if (isUndefined(features[name])) {
+        console.warn(
+            `LWC feature flag "${name}" is undefined. Possible reasons are that 1) it was misspelled or 2) it was removed from the @lwc/features package.`
+        );
+    }
     runtimeFlags[name] = value;
 }
 
-function setFeatureFlagForTest(name: string, value: FeatureFlag) {
-    if (process.env.NODE_ENV === 'production') {
-        // this method must not leak to production
-        throw new ReferenceError();
+// This function is added to the LWC API whitelist (for testing purposes) so we
+// add a check to make sure it is not invoked in production.
+function setFeatureFlagForTest(name: string, value: FeatureFlagValue) {
+    if (process.env.NODE_ENV !== 'production') {
+        if (isUndefined(features[name])) {
+            throw new Error(
+                `Feature flag "${name}" is undefined. Possible reasons are that 1) it was misspelled or 2) it was removed from the @lwc/features package.`
+            );
```

Comment:
We throw for tests but only log a warning when potentially in production (see `setFeatureFlag` above).